dbfit-java/sqlserver/src/main/java/dbfit/environment/SqlServerEnvironment.java

```diff
@@ -80,8 +81,28 @@ protected String parseCommandText(String commandText) {
         DbParameterAccessorsMapBuilder params = new DbParameterAccessorsMapBuilder();
 
         objname = objname.replaceAll("[^a-zA-Z0-9_.#$]", "");
+
+        String currentDatabaseName;
+        try (Statement s = currentConnection.createStatement()) {
+            ResultSet rs = s.executeQuery("SELECT DB_NAME() AS x");
+            rs.next();
+            currentDatabaseName = rs.getString(1);
+        }
+
+        String objectDatabaseName;
+        String[] objnameParts = objname.split("\\.");
+        if (objnameParts.length == 3) {
+            objectDatabaseName = objnameParts[0];
+        } else {
+            objectDatabaseName = currentDatabaseName;
+        }
+
         String bracketedName = enquoteAndJoin(objname.split("\\."), ".", "[", "]");
 
+        try (Statement s = currentConnection.createStatement()) {
+            s.execute("USE " + objectDatabaseName);
```

Comment:
Hmm, I think this has side effect: the current database is changed; and if something dies - it will remain in the new database.

What about just modifying the query above from `sys.columns` to something like `objectDatabaseName.sys.columns`?